components/BuyAssetForm/index.tsx

```diff
@@ -52,12 +52,17 @@ const BuyAssetForm = ({
     }
 
     try {
+      if (!currentUser) {
```

Comment:
Best to also check if there's currentUser.actor: `if (!currentUser || !currentUser.actor)`